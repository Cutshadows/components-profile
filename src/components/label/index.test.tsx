import { render, screen } from '@testing-library/react';
import Label from './index';

describe('Label component', () => {
	it('renders the correct text', () => {
		const labelText = 'Test Label';
		const attributes = { className: 'label', HTMLFormElement: 'label'};
		const {container} = render(<Label text={labelText}  {...attributes}/>);
		expect(container.querySelector('label')).toMatchSnapshot();
	});
	it('renders the label text correctly', () => {
		const labelText = 'Test Label';
		render(<Label text={labelText} />);
		expect(screen.getByText(labelText)).toBeInTheDocument();
	});

	it('applies the correct className prop', () => {
		const className = 'custom-label';
		const attributes = { className, HTMLFormElement: 'label'};
		render(<Label text="Test Label" {...attributes} />);
		expect(screen.getByText('Test Label')).toHaveClass(className);
	});

});