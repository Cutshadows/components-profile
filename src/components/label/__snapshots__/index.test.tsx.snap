// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label component > renders the correct text 1`] = `
<label
  class="label"
  htmlformelement="label"
>
  Test Label
</label>
`;
